polyfills/promise.js

```diff
@@ -127,7 +127,8 @@
 
     // %Promise% constructor.
     var cons = function Promise(executor) {
-        if (!new.target) { throw new Error('Promise must be called as a constructor'); }
+        if (!new.target) { throw new TypeError('Promise must be called as a constructor'); }
```

Comment:
Duktape supports `new.target` now?

Reply:
Yes, it was added in 2.2.0 release.

It's quite limited now though: it's always either undefined, or the current constructor function (if in a constructor call) because NewTarget and class support is not yet implemented. It's still useful to check for non-constructor calls.